### DISCUSSION TITLE
0.6.3 RC slower than 0.6.2 -> Buffer full after some time

### DISCUSSION THREAD

**Discussion by joule-ian** [QuantumEntangledAndy/neolink#301] [General]
Hello everybody,
I have tried 0.6.2 and 0.6.3 RC2 on my RPi5 with

_Core 2024.9.3
Supervisor 2024.09.1
Operating System 13.1
Frontend 20240909.1_

My observation is that 0.6.3 is slower. The delay is much bigger in 0.6.3 for my LUMUS.
Is there something I can change to speed up the RTSP Stream? Any parameters or features I can turn of in NeoLink config?

I also got a huge Log for running the RTSP Stream. My dealy is like 5 Minutes in RTSP-Stream and drifting. (With my other ONVIF Cam I have no delay and drift at all.)

Maybe some suggestions here? Auto purge or something?

Thanks a lot!

Log:

> [2024-09-23T09:32:13Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-23T09:32:16Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-23T09:32:17Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T08:03:49Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T08:03:49Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T11:00:08Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T11:00:08Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T11:26:16Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T11:26:16Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T11:26:20Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T11:26:20Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T14:46:39Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T14:46:39Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T14:46:43Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T14:46:43Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T14:46:47Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T14:46:47Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T14:46:51Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T14:46:51Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T14:46:55Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T14:46:55Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T14:46:59Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T14:46:59Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T14:46:59Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T14:46:59Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T15:26:48Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T15:26:48Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T15:26:52Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T15:26:52Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T15:26:56Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T15:26:56Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T15:27:00Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T15:27:00Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T15:27:04Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T15:27:04Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T15:27:08Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T15:27:08Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T15:27:12Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T15:27:12Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T15:27:16Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T15:27:16Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T15:27:20Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T15:27:20Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T15:27:24Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T15:27:24Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T15:27:28Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T15:27:28Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T15:27:32Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T15:27:32Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T15:27:36Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T15:27:36Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T15:27:40Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-24T15:27:40Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T20:37:59Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-24T20:37:59Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-25T06:25:31Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-25T06:25:31Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-09-25T06:25:35Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-09-25T06:25:35Z INFO  neolink::rtsp::stream] Buffer full on audsrc

**Comment by QuantumEntangledAndy** [comment]
Can you try the latest build on master? I've reworked the way connections are done to help with this already

**Reply by joule-ian** [reply]
Ok, I use neolink-latest addon in HA. The log file says 0.6.3-rc.3. Isn't that the latest build? Or how can I update?

    

> --- VERSIONS ---
>     add-on version: 0.0.1
>     [2024-09-25T07:07:56Z INFO neolink] Neolink https://github.com/QuantumEntangledAndy/neolink/commit/f15fdaefbcfe5cde3b0289c1e106cb85f6d3d39b release
>     neolink version: neolink 0.6.3-rc.3
>     neolink mode: rtsp
>     neolink log: info
>     ATTENTION: if you expected a newer Neolink version, please reinstall this Add-on!
>     --- Neolink ---
>     [2024-09-25T07:07:56Z INFO neolink] Neolink https://github.com/QuantumEntangledAndy/neolink/commit/f15fdaefbcfe5cde3b0289c1e106cb85f6d3d39b release
>     [2024-09-25T07:07:56Z INFO neolink::rtsp] Starting RTSP Server at 0.0.0.0:8554
>     [2024-09-25T07:07:56Z INFO neolink::rtsp] Lumus: Rtsp Starting
>     [2024-09-25T07:07:56Z INFO neolink::utils] Lumus: Connecting to camera at Address: 192.168.2.41, UID: 952700079XXXXXXX
>     [2024-09-25T07:07:56Z INFO neolink_core::bc_protocol] Lumus: Trying TCP discovery
>     [2024-09-25T07:07:56Z INFO neolink_core::bc_protocol] Lumus: TCP Discovery success at 192.168.2.41:9000
>     [2024-09-25T07:07:56Z INFO neolink::utils] Lumus: Logging in
>     [2024-09-25T07:07:57Z INFO neolink::utils] Lumus: Connected and logged in
>     [2024-09-25T07:07:59Z INFO neolink::common::camthread] Lumus: Camera time is already set: 2024-09-25 9:07:59.0 -01:00:00
>     [2024-09-25T07:08:01Z INFO neolink::common::neocam] Lumus: Model Reolink Lumus
>     [2024-09-25T07:08:01Z INFO neolink::common::neocam] Lumus: Firmware Version v3.1.0.3126_2401092472
>     [2024-09-25T07:08:01Z INFO neolink::rtsp::stream] Lumus: Available at /Lumus/sub, /Lumus/Sub, /Lumus/subStream, /Lumus/SubStream, /Lumus/Substream, /Lumus/substream, /Lumus

**Comment by QuantumEntangledAndy** [comment]
Some builds are [here](https://github.com/QuantumEntangledAndy/neolink/actions/runs/10233976452)

**Comment by joule-ian** [comment]
Hey there,

still getting these errors. After some time, now with updates, restarts and observation.

> [2024-11-07T20:20:02Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-11-07T20:20:06Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-11-07T20:20:06Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-11-07T20:20:10Z INFO  neolink::rtsp::stream] Buffer full on audsrc
> [2024-11-07T20:20:10Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-11-07T20:20:10Z INFO  neolink::rtsp::stream] Buffer full on vidsrc
> [2024-11-07T20:20:10Z INFO  neolink::rtsp::stream] Buffer full on audsrc

I changed the Port to and from 8556 back to standard 8554 and thought it disappeared, but errors still happening. Also massive drifting up to 5 minutes are normal after some time. 0.6.2 was more reliable on that.
